eng/common/scripts/Package-Properties.ps1

```diff
@@ -112,28 +114,38 @@ class PackageProps {
         return $null
     }
 
-    [void]InitializeCIArtifacts() {
-        if (-not $env:SYSTEM_TEAMPROJECTID  -and -not $env:GITHUB_ACTIONS) {
-            return
-        }
-
+    [PSCustomObject]GetCIYmlForArtifact() {
         $RepoRoot = Resolve-Path (Join-Path $PSScriptRoot ".." ".." "..")
 
         $ciFolderPath = Join-Path -Path $RepoRoot -ChildPath (Join-Path "sdk" $this.ServiceDirectory)
         $ciFiles = Get-ChildItem -Path $ciFolderPath -Filter "ci*.yml" -File
+        $ciArtifactResult = $null
+
+        foreach ($ciFile in $ciFiles) {
+            $ciArtifactResult = $this.ParseYmlForArtifact($ciFile.FullName)
```

Comment:
Just reading through the current and old code, are we getting lucky that we resolve ci.yml before ci.<variant>.yml and/or how do we resolve those differences? I think it's fair for us to require anything go in ci.yml (or a variant if no ci.yml exists) if you want it to be processed by the auto-pr pipeline.

Reply:
> are we getting lucky that we resolve ci.yml before ci.variant.yml and/or how do we resolve those difference

No, because we're looking _for the artifact in the artifactlist for the ci.yml_. That is our "yes this is the right ci.yml." This logic falls through if the artifact belongs to both, but I believe that's in error if the team has split their `ci.yml` into `ci.data.yml` and `ci.mgmt.yml`.

Reply:
I see where you're coming from though. This needs further refactoring to make it work the way we expect.

Reply:
Ok I'm going to resolve this comment chain, the format of this is going to be much more obvious now, even though I need to integration test a bit.